### PR TITLE
Create oauth2 token storage model…

### LINKED ISSUE #2103: Create Oauth2TokenStorage model
## Suggested Feature / Enhancement
- This new model is to store oauth2 (openid? or google) credentials

## Benefits of implementing the feature/enhancement
- Allow us to easily and efficiently migrate from deprecated oauth2client credentials

Part of #1992

### PR DESCRIPTION
**Pull request by WinnyTroy**
Create oauth2 token storage model for google auth credentials persistent storage

### Side effects of implementing this change
- New migrations for an empty db table 

Closes #2103

### CODE REVIEW

File: onadata/apps/api/models/oauthtokenstorage.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+"""
+Oauth2 token storage model class
+"""
+
+from django.db import models
+from jsonfield import JSONField
+from django.conf import settings
+
+
+class OauthStorageModel(models.Model):
```
Comment (DavisRayM) [inline onadata/apps/api/models/oauthtokenstorage.py:11]: Is the plan for this model to be used to store different OAuth credentials? Not only google? MIght be nice to have an easy way to differentiate which provider the credentials are for i.e google, microsoft, e.t.c

Comment (WinnyTroy) [reply onadata/apps/api/models/oauthtokenstorage.py:11]: At the moment, its mainly for the google Oauth2, but any other Oauth2 provider credentials may possibly also be persisted within this model.
How would you suggest we differentiat the provider?, include a separate third text field for oauth provider? - I dont see why not

File: onadata/apps/api/migrations/0009_oauthstoragemodel_provider.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.23 on 2021-07-28 13:49
+
```
Comment (DavisRayM) [inline onadata/apps/api/migrations/0009_oauthstoragemodel_provider.py:2]: Lets combine the migrations in this PR into one migration

**Comment by denniswambua** [comment]
@WinnyTroy Do we need a script to migrate the old tokens to this new token storage?

File: onadata/apps/api/migrations/0009_oauthstoragemodel_provider.py (hunk 2 above)
Comment (WinnyTroy) [reply onadata/apps/api/migrations/0009_oauthstoragemodel_provider.py]: done

**Comment by WinnyTroy** [comment]
@denniswambua populating this model will be handled in [this](https://github.com/onaio/onadata/issues/2104) separate issue.

**Comment by DavisRayM** [comment]
Superseded by #2236

**Pull request closed without merging**